src/makeInfinityAdder.js

```diff
@@ -6,7 +6,28 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let adderResult = 0;
+  let counter = 0;
+
+  return function adder(a, b = adderResult) {
+    counter++;
+
+    if (!a && counter === 1) {
+      adderResult = 0;
+      return adderResult
+    }
+
+    if (a) {
```

Comment:
Test is failing when 0 is passed, but instead the 0 should be added, and you should return function until `a` is undefined